fermenter-mda-test-utils/src/main/java/org/technologybrewery/fermenter/mda/test/utils/TestUtil.java

```diff
@@ -0,0 +1,82 @@
+package org.technologybrewery.fermenter.mda.test.utils;
+
+import org.technologybrewery.fermenter.mda.GenerateSourcesHelper;
+import org.technologybrewery.fermenter.mda.element.ExpandedProfile;
+import org.technologybrewery.fermenter.mda.element.Target;
+
+import java.io.*;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class TestUtil {
+
+
+
+    public static Map<String, ExpandedProfile> readProfilesFile(File profilesFile) throws IOException {
+        Map<String, ExpandedProfile> profilesMap = new HashMap<>();
+        InputStream profileStream = new FileInputStream(profilesFile);
+        profilesMap = GenerateSourcesHelper.loadProfiles(profileStream, profilesMap);
+
+        return profilesMap;
+    }
+
+    public static Map<String, Target> readTargetsFile(File targetsFile) throws IOException {
+        Map<String, Target> targetsMap = new HashMap<>();
+        InputStream targetStream = new FileInputStream(targetsFile);
+        targetsMap = GenerateSourcesHelper.loadTargets(targetStream, targetsMap);
+
+        return targetsMap;
+    }
+
+    public static void dereference(Map<String, ExpandedProfile> profileMap,  Map<String, Target> targetMap)
+    {
+        for (ExpandedProfile p : profileMap.values()) {
+            p.dereference(profileMap, targetMap);
+        }
+    }
+
+    public static boolean isProfileContainsAllSpecifiedTargetReferences(ExpandedProfile profile, List<String> targetReferences)
+    {
+        int targetContained = 0;
+        for(String targetRef : targetReferences)
+        {
+            if(profile.getTargets().stream().anyMatch(target -> target.getName().equals(targetRef)))
+            {
+                targetContained++;
+            }
+        }
+
+        return targetContained == targetReferences.size();
+    }
+
+    public static boolean isProfileContainsAllSpecifiedProfileReferences(ExpandedProfile profile, List<String> profileReferences)
+    {
+        int profileContained = 0;
+        for(String profileRef : profileReferences)
+        {
+            if(profile.getReferencedProfiles().stream().anyMatch(p -> p.getName().equals(profileRef)))
+            {
+                profileContained++;
+            }
+        }
+
+        return profileContained == profileReferences.size();
+    }
+
+    public static boolean isTemplateFileExist(String fullTemplateFilePath){
```

Comment:
A: It might make more sense for these methods to directly take the `Target` object.

Reply:
updated param!